Sources/SwiftKafka/KafkaConsumer.swift

```diff
@@ -0,0 +1,418 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+import struct Foundation.UUID // TODO: can we avoid this Foundation import??
+import Logging
+import NIOCore
+
+/// `NIOAsyncSequenceProducerDelegate` implementation handling backpressure for ``KafkaConsumer``.
+actor ConsumerMessagesAsyncSequenceDelegate: NIOAsyncSequenceProducerDelegate {
+    weak var consumer: KafkaConsumer?
+
+    // TODO: is it ok to block here -> we can make poll non-async and get rid of the task here
+    nonisolated func produceMore() {
```

Comment:
Is it ok if `producerMore()` is blocking? We could then get rid of the `Task` etc.